.github/workflows/docker.yml

```diff
@@ -0,0 +1,57 @@
+name: "Publish Docker Image"
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+    tags:
+      - "v*"   push:
+    branches:
+      - main
+    tags:
+      - "v*"
+
+jobs:
+  build-docker-image:
+    name: "Build Docker Image"
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@main
+      - name: Magic Nix Cache
+        uses: DeterminateSystems/magic-nix-cache-action@main
+
+      - name: Build Docker Image
+        run: nix build .#docker-nix --print-build-logs
+
+      - name: Load & Tag Built Docker Image
+        run: |
+          docker load -i result ; docker images &&
```

Comment:
`docker images` is there for debugging purpose, right ? maybe remove it /

Reply:
oh yeah, goodo ne